cranelift/codegen/src/isa/riscv64/inst.isle

```diff
@@ -2850,15 +2846,16 @@
   (gen_select_reg (IntCC.SignedGreaterThan) x y x y))
 
 
-(decl gen_trapif (XReg TrapCode) InstOutput)
-(rule
-  (gen_trapif test trap_code)
-  (side_effect (SideEffectNoResult.Inst (MInst.TrapIf test trap_code))))
+;; Builds an instruction sequence that traps if the comparision succeeds.
+(decl gen_trapif (IntCC XReg XReg TrapCode) InstOutput)
+(rule (gen_trapif cc a b trap_code)
+  (side_effect (SideEffectNoResult.Inst (MInst.TrapIf a b cc trap_code))))
+
+;; Builds an instruction sequence that traps if the input is zero.
+(decl gen_trapz (XReg TrapCode) InstOutput)
+(rule (gen_trapz test trap_code)
+  (gen_trapif (IntCC.NotEqual) test (zero_reg) trap_code))
```

Comment:
This is confusing me a bit as `trapif` seems to trap if the condition succeeds, so the condition here is that if the input is not equal to zero this is trapping, asserting the register is zero. The function, however, is intended to trap if the register is zero.

Should the helper perhaps be `trapnz`?

Reply:
Yeah, looks like I got a bit confused here. Thanks for catching this!